Index synchronization fixes.

srksag opened the pull request:
Sam,
I found an issue with index synchronization logic. There is a problem with the logic when there are subtypes defined for a template. I've fixed that issue and also cleaned up some code. 

I also feel that we may need to run the index cleanup scripts once when we deploy these changes to uat and production as the index definitions might've been out of sync in these environments if we synchronize definitions from different branches. e.g. I don't see index on email field on customer table in my local database, but this is defined in schema.json object, but I don;t see this problem in our CI database. We can even delete all the records from haven_schema1 table.

Sorry I should've got this error earlier or at least realized when you were facing the issues, but for some reason, when I was running the unit tests, things were working fine.

Can you please accept this pull request.


Thanks,
Ravi

selsamman commented:
Will do.  I was running into sync problems with Amorphic tests so have not
published this yet.  Hopefully this will take care of that.

I am thinking that maybe we just change the name of the schema to something
more generic like index_schema_history and then that will effectively
"delete" the old history.  I can take care of that when merging the PR.

....Sam

On Tue, May 17, 2016 at 12:48 PM, srksag <notifications@github.com> wrote:

> Sam,
> I found an issue with index synchronization logic. There is a problem with
> the logic when there are subtypes defined for a template. I've fixed that
> issue and also cleaned up some code.
>
> I also feel that we may need to run the index cleanup scripts once when we
> deploy these changes to uat and production as the index definitions
> might've been out of sync in these environments if we synchronize
> definitions from different branches. e.g. I don't see index on email field
> on customer table in my local database, but this is defined in schema.json
> object, but I don;t see this problem in our CI database. We can even delete
> all the records from haven_schema1 table.
>
> Sorry I should've got this error earlier or at least realized when you
> were facing the issues, but for some reason, when I was running the unit
> tests, things were working fine.
>
> Can you please accept this pull request.
>
> Thanks,
> Ravi
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/selsamman/persistor/pull/19
> Commit Summary
>
>    - index synchronization fixes
>    - Merge branch 'master' of https://github.com/selsamman/persistor
>
> File Changes
>
>    - *M* lib/knex/db.js
>    <https://github.com/selsamman/persistor/pull/19/files#diff-0> (47)
>
> Patch Links:
>
>    - https://github.com/selsamman/persistor/pull/19.patch
>    - https://github.com/selsamman/persistor/pull/19.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/selsamman/persistor/pull/19>
>

srksag commented:
Thanks Sam,Are you facing any issues with the scenario "sam looks good on refresh" (line #684) where it checks for expect(customer.secondaryAddresses[0].returnedMail.length).to.equal(2). This scenario is failing intermittently, This is nothing to do our changes. 
I'm including the table name change as well and I'm going to submit a new pull request.
Thanks,Ravi 

    On Tuesday, May 17, 2016 3:50 PM, selsamman <notifications@github.com> wrote:
 

 Will do. I was running into sync problems with Amorphic tests so have not
published this yet. Hopefully this will take care of that.

I am thinking that maybe we just change the name of the schema to something
more generic like index_schema_history and then that will effectively
"delete" the old history. I can take care of that when merging the PR.

....Sam

On Tue, May 17, 2016 at 12:48 PM, srksag <notifications@github.com> wrote:

> Sam,
> I found an issue with index synchronization logic. There is a problem with
> the logic when there are subtypes defined for a template. I've fixed that
> issue and also cleaned up some code.
>
> I also feel that we may need to run the index cleanup scripts once when we
> deploy these changes to uat and production as the index definitions
> might've been out of sync in these environments if we synchronize
> definitions from different branches. e.g. I don't see index on email field
> on customer table in my local database, but this is defined in schema.json
> object, but I don;t see this problem in our CI database. We can even delete
> all the records from haven_schema1 table.
>
> Sorry I should've got this error earlier or at least realized when you
> were facing the issues, but for some reason, when I was running the unit
> tests, things were working fine.
>
> Can you please accept this pull request.
>
> Thanks,
> Ravi
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/selsamman/persistor/pull/19
> Commit Summary
>
> - index synchronization fixes
> - Merge branch 'master' of https://github.com/selsamman/persistor
>
> File Changes
>
> - *M* lib/knex/db.js
> <https://github.com/selsamman/persistor/pull/19/files#diff-0> (47)
>
> Patch Links:
>
> - https://github.com/selsamman/persistor/pull/19.patch
> - https://github.com/selsamman/persistor/pull/19.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/selsamman/persistor/pull/19>
>
—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub

srksag commented:
going to submit a new request
